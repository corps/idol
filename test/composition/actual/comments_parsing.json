{"comments_parsing":{"module_name":"comments_parsing","types_by_name":{"TypeB":{"fields":{"c":{"field_name":"c","tags":["description:And this would","description:as well"],"type_struct":{"literal":null,"primitive_type":"int","reference":{"module_name":"","qualified_name":"","type_name":""},"struct_kind":"Scalar"},"optional":false}},"is_a":null,"named":{"module_name":"comments_parsing","qualified_name":"comments_parsing.TypeB","type_name":"TypeB"},"options":[],"tags":["description:Obviously this one","description:This would be added? to the type comments"]},"TypeA":{"fields":{"a":{"field_name":"a","tags":["description:This field would receive comments","description:Multiline ones too","description:Stripping and spacing"],"type_struct":{"literal":null,"primitive_type":"string","reference":{"module_name":"","qualified_name":"","type_name":""},"struct_kind":"Scalar"},"optional":false},"b":{"field_name":"b","tags":["description:But this one would"],"type_struct":{"literal":null,"primitive_type":"int","reference":{"module_name":"","qualified_name":"","type_name":""},"struct_kind":"Scalar"},"optional":false}},"is_a":null,"named":{"module_name":"comments_parsing","qualified_name":"comments_parsing.TypeA","type_name":"TypeA"},"options":[],"tags":["abc","description:This type would receive comments"]}},"types_dependency_ordering":["TypeB","TypeA"]}}
