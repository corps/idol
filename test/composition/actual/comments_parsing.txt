=== module comments_parsing
comments_parsing.TypeB: {
  c: int # And this would  as well
} # Obviously this one  This would be added? to the type comments
comments_parsing.TypeA: {
  a: string # This field would receive comments  Multiline ones too  Stripping and spacing
  b: int # But this one would
} # This type would receive comments
