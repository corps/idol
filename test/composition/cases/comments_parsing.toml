# This type would receive comments
[TypeA]


# This field would receive comments
# Multiline ones too
#Stripping and spacing  
fields.a = "string"

# This would be not become a field comment
tags = ["abc"]

# But this one would
fields.b = "int"

# Obviously this one

[TypeB]
# Comment would be ignored
tags = []


# This would be added? to the type comments
[TypeB.fields]
# And this would
# as well
c = "int"

# This would be discarded
