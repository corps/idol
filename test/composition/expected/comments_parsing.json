{
  "comments_parsing": {
    "module_name": "comments_parsing",
    "types_by_name": {
      "TypeA": {
        "fields": {
          "a": {
            "field_name": "a",
            "optional": false,
            "tags": [
              "description:This field would receive comments",
              "description:Multiline ones too",
              "description:Stripping and spacing"
            ],
            "type_struct": {
              "literal": null,
              "primitive_type": "string",
              "reference": {
                "module_name": "",
                "qualified_name": "",
                "type_name": ""
              },
              "struct_kind": "Scalar"
            }
          },
          "b": {
            "field_name": "b",
            "optional": false,
            "tags": [
              "description:But this one would"
            ],
            "type_struct": {
              "literal": null,
              "primitive_type": "int",
              "reference": {
                "module_name": "",
                "qualified_name": "",
                "type_name": ""
              },
              "struct_kind": "Scalar"
            }
          }
        },
        "is_a": null,
        "named": {
          "module_name": "comments_parsing",
          "qualified_name": "comments_parsing.TypeA",
          "type_name": "TypeA"
        },
        "options": [],
        "tags": [
          "abc",
          "description:This type would receive comments"
        ]
      },
      "TypeB": {
        "fields": {
          "c": {
            "field_name": "c",
            "optional": false,
            "tags": [
              "description:And this would",
              "description:as well"
            ],
            "type_struct": {
              "literal": null,
              "primitive_type": "int",
              "reference": {
                "module_name": "",
                "qualified_name": "",
                "type_name": ""
              },
              "struct_kind": "Scalar"
            }
          }
        },
        "is_a": null,
        "named": {
          "module_name": "comments_parsing",
          "qualified_name": "comments_parsing.TypeB",
          "type_name": "TypeB"
        },
        "options": [],
        "tags": [
          "description:Obviously this one",
          "description:This would be added? to the type comments"
        ]
      }
    },
    "types_dependency_ordering": [
      "TypeB",
      "TypeA"
    ]
  }
}
